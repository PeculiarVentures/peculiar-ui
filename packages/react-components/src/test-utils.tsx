import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
