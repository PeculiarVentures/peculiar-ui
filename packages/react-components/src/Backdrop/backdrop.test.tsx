import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Backdrop } from './index';

describe('<Backdrop />', () => {
  describe('Backdrop render variants', () => {
    it('should render as default', () => {
      const { asFragment } = render(
        <Backdrop open />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
