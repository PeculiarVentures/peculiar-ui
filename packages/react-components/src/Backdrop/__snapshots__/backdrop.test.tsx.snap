// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> Backdrop render variants should render as default 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
}

<div
    aria-hidden="true"
    class="emotion-0"
    style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
</DocumentFragment>
`;
